session/session.go

```diff
@@ -2618,37 +2630,31 @@ func (s *session) loadCommonGlobalVariablesIfNeeded() error {
 	}
 
 	var err error
-	// Use GlobalVariableCache if TiDB just loaded global variables within 2 second ago.
-	// When a lot of connections connect to TiDB simultaneously, it can protect TiKV meta region from overload.
-	gvc := domain.GetDomain(s).GetGlobalVarsCache()
-	loadFunc := func() ([]chunk.Row, []*ast.ResultField, error) {
-		vars := append(make([]string, 0, len(builtinGlobalVariable)+len(variable.PluginVarNames)), builtinGlobalVariable...)
-		if len(variable.PluginVarNames) > 0 {
-			vars = append(vars, variable.PluginVarNames...)
-		}
 
-		stmt, err := s.ParseWithParams(context.TODO(), "select HIGH_PRIORITY * from mysql.global_variables where variable_name in (%?) order by VARIABLE_NAME", vars)
-		if err != nil {
-			return nil, nil, errors.Trace(err)
-		}
+	vars.CommonGlobalLoaded = true
```

Comment:
`CommonGlobalLoaded` could be removed, since you already have the `loadIfNeeded` logic in the `sysvarCache`. I am OK if you want to do that in the next PR.

And possibly, `loadCommonGlobalVariablesIfNeeded` could be removed. It should be something like `-> sysvarCache ->  persistence`. But, in fact, there is an extra deep copy `sessionVars -> sysvarCache > persistence`.